### PR TITLE
Allow events to use config IP

### PR DESCRIPTION
**Pull request by lwis**

### CODE REVIEW

**Comment by lawrenceakka** [comment]
Hi,

Thanks for this but please can you explain what it is for? What particular problem are you trying to solve?

**Comment by lwis** [comment]
@lawrenceakka this was created off the back of home-assistant/home-assistant#4177, essentially when running in Docker containers the host IP isn't obtainable/exposed - so when SoCo binds and advertises itself for the event stream to other speakers we end up in a situation where it's sending out the IP of the Docker container, which the speakers cannot connect to.

File: soco/config.py [hunk 1]
```diff
@@ -31,6 +31,16 @@
 """
 
 
+EVENT_LISTENER_IP = None
+"""The IP on which the event listener listens.
+
+The default is None. If unset it will detect.
```
Comment (lawrenceakka) [inline soco/config.py]: Please change to "The default of None means that the relevant IP address will be detected automatically"

File: soco/events.py [hunk 2]
```diff
@@ -317,11 +317,16 @@ def start(self, any_zone):
         # Sonos net, see http://stackoverflow.com/q/166506
         with self._start_lock:
             if not self.is_running:
-                temp_sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-                temp_sock.connect((any_zone.ip_address,
-                                   config.EVENT_LISTENER_PORT))
-                ip_address = temp_sock.getsockname()[0]
-                temp_sock.close()
+                if config.EVENT_LISTENER_IP:
```
Comment (lawrenceakka) [inline soco/events.py]: Please add a comment just before this line (We like comments 😄 ):

#Use configured IP address if there is one, else detect automatically

**Comment by lwis** [comment]
@lawrenceakka done, cheers bud!

**Comment by lawrenceakka** [comment]
No other comments received, so I'm committing this. Thanks very much.

**Pull request merged**